### PR TITLE
feat(#2148): enable JTCOP.RuleAllTestsHaveProductionClass check for eo-maven-plugin

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Enable `JTCOP.RuleAllTestsHaveProductionClass` check for `eo-maven-plugin`.

Closes: #2148

---

## PR-Codex overview
This PR removes some test files, adds and modifies some test classes, and updates the `pom.xml` file. Notable changes include:

- Renaming `DcsJsonTest` to `DcsDepgraphTest`
- Adding tests for `LogFormatTest` and `CargoCondition`
- Adding `Online` class to `OnlineCondition`
- Adding `CaptureLogs` class and `CaptureLogsExtension` class to capture log messages in tests.

> The following files were skipped due to too many changes: `eo-maven-plugin/src/test/java/org/eolang/maven/log/CaptureLogs.java`, `eo-maven-plugin/src/test/java/org/eolang/maven/SnippetTest.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@Graur Could you have a look, please?

**Comment by volodya-lombrozo** [comment]
@Graur I have fixed the problems with qulice and codacy. Could you have a look, please?

File: eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ void assemblesTogether(@TempDir final Path temp) throws Exception {
                     )
                 )
             ),
-            Matchers.is(new Online().value())
+            Matchers.is(true)
```
Comment (Graur) [inline eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java:89]: @volodya-lombrozo Why do we need this change?

File: eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java [hunk 2]
```diff
@@ -150,7 +150,7 @@ void assemblesNotFailWithFailOnErrorFlag(@TempDir final Path temp) throws Except
...
```
Comment (Graur) [inline eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java:153]: @volodya-lombrozo Why do we need this change?

**Review by Graur — CHANGES_REQUESTED** [review]

File: eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java (hunk 1 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java:89]: @Graur We already have `@ExtendWith(OnlineCondition.class)` statement which is declared for `AssembleMojoTest`. It means, if `new Online().value() == false`, then the tests won't be invoked. Thus, we can just put the `true`. Moreover, by removing `new Online().value()` we can hide this class inside `OnlineCondition` class.

File: eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java (hunk 2 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java:153]: @Graur The same as [here](https://github.com/objectionary/eo/pull/2180#discussion_r1243831517).

File: eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java (hunk 1 above)
Comment (Graur) [reply eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java:89]: @volodya-lombrozo Thanks!

**Review by Graur — APPROVED** [review]
@volodya-lombrozo Looks good to me
@yegor256 Please have a look

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you merge that changes, please?

**Pull request merged**